shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/widget/presentation/StudyPlanWidgetReducer.kt

```diff
@@ -188,10 +188,27 @@ class StudyPlanWidgetReducer : StateReducer<State, Message, Action> {
                 val activitiesIdsToRemove = activitiesIds - message.activities.map { it.id }.toSet()
                 activitiesIdsToRemove.forEach { remove(it) }
             },
-            studyPlanSections = state.studyPlanSections.update(message.sectionId) { sectionInfo ->
-                sectionInfo.copy(contentStatus = StudyPlanWidgetFeature.ContentStatus.LOADED)
+            // ALTAPPS-786: We should hide sections without available activities to avoid blocking study plan
+            studyPlanSections = if (message.activities.isNotEmpty()) {
```

Comment:
Let's change the condition so that just after the comment we will see a true statement:
```kotlin
// ALTAPPS-786: We should hide sections without available activities to avoid blocking study plan
studyPlanSections = if (message.activities.isEmpty()) {
    state.studyPlanSections.toMutableMap().apply {
        remove(message.sectionId)
    }
} else {
    state.studyPlanSections.update(message.sectionId) { sectionInfo ->
        sectionInfo.copy(contentStatus = StudyPlanWidgetFeature.ContentStatus.LOADED)
    }
}
```